File: VueInterface/app-ui/src/models/AppSettings.ts
export class AppSettings{
    darkTheme?: Boolean;
    playHotkey?: KeyCombination;
    recordHotkey?: KeyCombination;
    topHotkey?: KeyCombination;
    typeSpeed?: Number;
    language?: String;
    version?: string = "2.0.0";
    port?: number;
    constructor(){
        this.playHotkey = new KeyCombination();
        this.recordHotkey = new KeyCombination();
        this.topHotkey = new KeyCombination();
    }
}

export class KeyCombination{
    ctrlKey?: Boolean;
    altKey?: Boolean;
    shiftKey?: Boolean;
    key?: String;
    constructor(){
        this.ctrlKey = false;
        this.altKey= false;
        this.shiftKey=false;
        this.key='';
    }

    static clear(src: KeyCombination){
        src.ctrlKey=false;
        src.altKey=false;
        src.shiftKey=false;
        src.key='';
    }

    static compare(src: KeyCombination, that: KeyCombination){
        return src.altKey == that.altKey &&
        src.ctrlKey == that.ctrlKey &&
        src.shiftKey == that.shiftKey &&
        src.key == that.key;
    }
}

File: Autoquit2/Models/AppSettings.cs
using Autoquit2._App_Data;
using System.Windows.Forms;

namespace Autoquit2.Models {
    public class AppSettings {
        public bool DarkTheme { get; set; } = false;
        public KeyCombination PlayHotkey { get; set; } = new KeyCombination() {
            Key =""
        };
        public KeyCombination RecordHotkey { get; set; } = new KeyCombination() {
            Key = ""
        };
        public KeyCombination TopHotkey { get; set; } = new KeyCombination() {
            Key = ""
        };
        public bool ShowMouseCoordinate { get; set; } = false;
        public bool AutoMerge { get; set; } = true;
        public int TypeSpeed { get; set; } = 10;
        public string Language { get; set; } = "en-US";
        public int Port { get; set; } = 7709;
        public string Version { get {
                return Constant.AppVersion;
            } 
        }
    }
}
